### PR TITLE
Externalize JavaScript: domain dashboard

### PR DESCRIPTION
**Pull request by orangejenny**
@esoergel

### CODE REVIEW

File: corehq/apps/dashboard/static/dashboard/js/dashboard_domain.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+hqDefine('dashboard/js/dashboard_domain.js', function() {
+    'use strict';
+    var initial_page_data = hqImport('hqwebapp/js/initial_page_data.js').get;
+    var dashboardApp = window.angular.module('dashboardApp', [
+        'ui.bootstrap',
+        'ngResource',
+        'ngRoute',
+        'ng.django.rmi',
+        'hq.dashboard'
```
Comment (dimagimon) [inline corehq/apps/dashboard/static/dashboard/js/dashboard_domain.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/hqwebapp/views.py [hunk 2]
```diff
@@ -1194,3 +1195,18 @@ def couch_doc_counts(request, domain):
...
+    # Add the output of djng_current_rmi to view context, which requires having
+    # the rest of the context, specifically context['view'], available.
+    # See https://github.com/jrief/django-angular/blob/master/djng/templatetags/djng_tags.py
+    def get_context_data(self, **kwargs):
+        context = super(HQJSONResponseMixin, self).get_context_data(**kwargs)
+        from djangular.templatetags.djangular_tags import djng_current_rmi
+        context['djng_current_rmi'] = json.loads(djng_current_rmi(context))
+        return context
+
+
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: W391 blank line at end of file

**Comment by orangejenny** [comment]
or @millerdev

File: corehq/apps/hqwebapp/views.py [hunk 3]
```diff
@@ -1194,3 +1195,16 @@ def couch_doc_counts(request, domain):
         cls.__name__: get_doc_count_in_domain_by_class(domain, cls, start, end)
         for cls in [CommCareCase, XFormInstance]
     })
+
+
+# Use instead of djangular's base JSONResponseMixin
+# Adds djng_current_rmi to view context
+class HQJSONResponseMixin(JSONResponseMixin):
```
Comment (esoergel) [inline corehq/apps/hqwebapp/views.py]: :+1:

**Review by esoergel — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Pull request merged**